pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+            (schema_version[0], schema_version[1], schema_version[2],
+             self._mof_zip_bn)
+        schema_mof_bld_name = self._dmtf_schema_version + '.mof'
+
+        # This is object level variable only because of tests
+        self._mof_zip_filename = os.path.join(self.schema_root_dir,
+                                              self._mof_zip_bn)
+        self._schema_mof_filename = os.path.join(self.schema_mof_dir,
+                                                 schema_mof_bld_name)
+
+        self.verbose = verbose
+
+        # Setup the local copy of the DMTF CIM schema which may include creating
+        # the schema directory, downloading the schema zip file from the DMTF
+        # web site, and expanding the schema into a subdirectory of the schema
+        # directory.
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_version(self):
+        """
+        :func:`py:tuple`: of 3 integers with major version, minor version, and
+        update version of the DMTF Schema installed. Ex (2, 49, 0) defines DMTF
+        schema version 2.49.0.
+        """
+        return self._schema_version
+
+    @property
+    def schema_version_str(self):
+        """
+        :term:`string`: with the DMTF CIM schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_version[0], self.schema_version[1],
+                             self.schema_version[2])
+
+    @property
+    def schema_root_dir(self):
+        """
+        :term:`string`: the directory in which the DMTF CIM Schema
+        MOF is downloaded and expanded.
+        """
+        return self._schema_root_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string`: Path name of the directory in which the DMTF CIM Schema
+        MOF files are expanded from the downloaded zip file. This includes the
+        expansion of the individual schema MOF files. This property is useful
+        since it can be used as the MOF compiler search path for compiling
+        classes in the DMTF CIM schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string`: the path for the DMTF CIM schema MOF top level file
+        which includes the pragmas that define all of the files in the DMTF
+        schema. This is the file that the compiler uses if the complete
+        schema is to be compiled. This filename is of the general form::
+
+            cim_schema<schema_version>.mof
+
+        For example::
+
+            cim_schema_2.49.0.mof
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object for human consumption.
+        """
+
+        return '%s(schema_version=%s, schema_root_dir=%s, url=%s)' % \
```

Comment:
Rename "schema_version=" to the property that is actually shown, i.e. "schema_version_str=".

Reply:
Agree with second.  Added schema_zip_file.  However, we are showing the tuple form here and that is schema_version, not schema_version_str.

DISCUSSION

Reply:
If we show the version tuple, that's fine. DONE